### ISSUE TITLE
Create a new android-component that uses the SDK

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/application-services#3514]
A new android component should consume the published Nimbus SDK module and define the public Kotlin API for consumers such as Fenix.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-1682)
┆Story Points: 5
┆Epic: Nimbus Fenix Client SDK
┆Sprint: SYNC - end 2020-10-08

**Comment by data-sync-user** [comment]
➤ Travis Long commented:

I've opened a PR to add a "Nimbus" component to A-C: https://github.com/mozilla-mobile/android-components/pull/8261, ( https://github.com/mozilla-mobile/android-components/pull/8261 )

Not planning on merging until after meeting with Vlad and Tarik to discuss Android implementation on Aug 28th, just in case there is something we need/want to add to the initial component PR. 

We should also discuss in that meeting in how we plan to deliver the nimbus-sdk, which I'm guessing will be through Mozilla maven?  This will need to be figured out so we can consume it in A-C.

**Comment by data-sync-user** [comment]
➤ Travis Long commented:

I've reached out to Mihai Tabara regarding getting the nimbus-sdk built on Taskcluster, which simplifies getting it published to the Mozilla Maven instance.  He said it should be fairly straightforward, but he would get back to me by the end of this week with any clarifications or questions.

**Comment by data-sync-user** [comment]
➤ Travis Long commented:

Mihai has provided a bug that outlines 2 approaches to take to get Nimbus-SDK deployed to the Mozilla Maven instance.  One requires more engineering effort and is the correct, automated, long-term solution.  The other option that we can use short-term in order to unblock us is a deployment script that we used for Glean.  Basically it's a manual operation but it would allow us to deploy to Maven now to unblock work on the Android-Component and Reference Browser.  See the bug for more information: https://bugzilla.mozilla.org/show_bug.cgi?id=1663212

**Comment by data-sync-user** [comment]
➤ Travis Long commented:

Looks like I am still blocked on this for the following reasons:

  * Waiting for publishing to Mozilla Maven in order to be able to consume this from A-C. 
  * Initialization of Viaduct in the Android Component

Another "nice to have" would be the ability to set the package-name through uniffi-bindgen, but probably not blocking the initial landing of the component.

**Comment by rfk** [comment]
> Waiting for publishing to Mozilla Maven in order to be able to consume this from A-C.

It's not clear to me what we need to do to unblock this. Would it help if we pulled nimbus into a megazord published by the appservices repo, which already has some publishing infra set up?

**Comment by rfk** [comment]
> Another "nice to have" would be the ability to set the package-name through uniffi-bindgen

The soon-to-be-released [uniffi 0.2.0](https://github.com/mozilla/uniffi-rs/pull/291) will support this, per the [examples in this PR](https://github.com/mozilla/uniffi-rs/pull/273/files#diff-e3557d313cd31a5eee096e109092168b).

**Comment by eoger** [comment]
`v0.2.0` is out.

**Comment by data-sync-user** [comment]
➤ Janet Dragojevic commented:

blocked on viadut 1779 and taskcluster build stuff.  We can immediately unblock by shipping a version of a-s with nimbus (ryan to create a new ticket)

**Comment by data-sync-user** [comment]
➤ Ryan Kelly commented:

> We can immediately unblock by shipping a version of a-s with nimbus (ryan to create a new ticket)

I filed https://jira.mozilla.com/browse/SYNC-1528 about resolving the publishing situation (but without assuming the ship-it-via-appservices is necessarily the correct approach)

**Comment by rfk** [comment]
This was done
